stdlib/synch.ysh

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env ysh
+
+module stdlib/synch || return 0
+
+############################
+### FIFO File Desriptors ###
+############################
+
+func fifoFdNew() {
+  # WARN: this section should be critical but for now it's not
+  # A solution may be retry on fail.
+  #====================
+  var fifo = $(mktemp -u)
```

Comment:
Functions will likely become **pure**, disallowing `$()`

https://oilshell.zulipchat.com/#narrow/stream/384942-language-design/topic/Why.20are.20functions.20pure.2C.20with.20IO.20type.3F

We may want to use a proc instead


```
var fifo
fifo-fd-new (&fifo)
```

Reply:
How do you return a non int value in a proc then? For now these have to be function as some functions returns non-int. And since purity is transitive the called function is also forced to be function.